lib/test.js

```diff
@@ -122,6 +122,15 @@ Test.prototype.end = function(fn){
   var server = this._server;
   var end = Request.prototype.end;
 
+  if (!fn) {
+    return new Promise(function (resolve, reject) {
```

Comment:
Be aware that the .pipe() method of Superagent expects .end() to return the instance itself and not a promise. IE. this breaks .pipe(). I managed to "promisify" .end() like shown below.

```javascript
var originalEnd = Test.prototype.end;

Test.prototype.end = function (callback) {
  var deferred = Promise.defer();

  originalEnd.call(this, function (err, res) {
    if (callback) {
      callback(err, res);
    }

    if (err) {
      deferred.reject(err);
    } else {
      deferred.resolve(res);
    }
  });

  this.then = deferred.promise.then.bind(deferred.promise);
  this.catch = deferred.promise.catch.bind(deferred.promise);

  return this;
};
```